### PR TITLE
Fix/na_values_GH59303

### LINKED ISSUE #59303: QST: Is this expected behavior when pd.read_csv() with na_values arguments?
### Research

- [X] I have searched the [[pandas] tag](https://stackoverflow.com/questions/tagged/pandas) on StackOverflow for similar questions.

- [X] I have asked my usage related question on [StackOverflow](https://stackoverflow.com).

### Link to question on StackOverflow

https://stackoverflow.com/questions/46397526/how-to-use-na-values-option-in-the-pd-read-csv-function

### Question about pandas

I have a simple csv file that looks like this:

```
x,y,z
a,-99,100
b,-99,200
c,-99.0,300
d,-99.0,400
```

and when I tried a few different na_values, I got different column y back:

```
import pandas as pd

df1 = pd.read_csv('test.csv', na_values={"y": -99})
print("df1 = \n", df1)

df2 = pd.read_csv('test.csv', na_values={"y": -99.0})
print("\ndf2 = \n", df2)

df3 = pd.read_csv('test.csv', na_values={"y": [-99.0, -99]})
print("\ndf3 = \n", df3)

df4 = pd.read_csv('test.csv', na_values={"y": [-99, -99.0]})
print("\ndf4 = \n", df4)
```

Results:

```
df1 = 
    x   y    z
0  a NaN  100
1  b NaN  200
2  c NaN  300
3  d NaN  400

df2 = 
    x     y    z
0  a -99.0  100
1  b -99.0  200
2  c   NaN  300
3  d   NaN  400

df3 = 
    x     y    z
0  a -99.0  100
1  b -99.0  200
2  c   NaN  300
3  d   NaN  400

df4 = 
    x   y    z
0  a NaN  100
1  b NaN  200
2  c NaN  300
3  d NaN  400
```

I'm not sure if this is a bug or it is by design, so just throwing out a general question here. Thank you!

Pandas version is 2.2.1, just in case needed.

### PR DESCRIPTION
**Pull request by sshu2017**
- [x] closes #59303 
- [x] Test added and passed: test_na_values_dict_without_dtype() in test_na_values.py 
- [x] All code checks passed.
- [x] No new type annotations.
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst`

### CODE REVIEW

File: pandas/tests/io/parser/test_na_values.py [hunk 1]
```diff
@@ -812,3 +812,23 @@ def test_bool_and_nan_to_float(all_parsers):
...
+-99.0"""
+    # this would FAIL BEFORE this fix
+    result_1 = parser.read_csv(StringIO(data), na_values={"A": [-99.0, -99]})
+    expected_1 = DataFrame.from_dict({"A": [np.nan, np.nan, np.nan, np.nan]})
+    tm.assert_frame_equal(result_1, expected_1)
+
+    # this would PASS even BEFORE this fix
+    result_2 = parser.read_csv(StringIO(data), na_values={"A": [-99, -99.0]})
+    expected_2 = DataFrame.from_dict({"A": [np.nan, np.nan, np.nan, np.nan]})
+    tm.assert_frame_equal(result_2, expected_2)
```
Comment (mroeschke) [inline pandas/tests/io/parser/test_na_values.py:834]: Could you use `@pytest.mark.parametrize("values", [[-99.0, -99], [-99, -99.0]])` to de-duplicate this?

File: doc/source/whatsnew/v3.0.0.rst [hunk 2]
```diff
@@ -706,6 +706,7 @@ Other
 - Bug in :meth:`Series.dt` methods in :class:`ArrowDtype` that were returning incorrect values. (:issue:`57355`)
 - Bug in :meth:`Series.rank` that doesn't preserve missing values for nullable integers when ``na_option='keep'``. (:issue:`56976`)
 - Bug in :meth:`Series.replace` and :meth:`DataFrame.replace` inconsistently replacing matching instances when ``regex=True`` and missing values are present. (:issue:`56599`)
+- Bug in :meth:`_clean_na_values` in :class:`TextFileReader` that was not properly handling ``na_values`` when it is a list of strings. (:issue:`59303`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:709]: Can you put this under the `I/O` section and not mention private objects such as `_clean_na_values` and `TextFileReader`?

File: pandas/tests/io/parser/test_na_values.py (hunk 1 above)
Comment (sshu2017) [reply pandas/tests/io/parser/test_na_values.py]: Implemented as suggested.

File: doc/source/whatsnew/v3.0.0.rst (hunk 2 above)
Comment (sshu2017) [reply doc/source/whatsnew/v3.0.0.rst]: Implemented as suggested. Thank you @mroeschke.

**Comment by sshu2017** [comment]
I think the two comments on the PR are now resolved, but I don't understand why some of the Unit Tests fail. I looked into the details of the fails, but they seem to be not related to this PR - If that is the case, please kindly suggest what I should do next, if there's anything needed from my end. Thank you!

File: pandas/tests/io/parser/test_na_values.py [hunk 3]
```diff
@@ -812,3 +812,32 @@ def test_bool_and_nan_to_float(all_parsers):
     result = parser.read_csv(StringIO(data), dtype="float")
     expected = DataFrame.from_dict({"0": [np.nan, 1.0, 0.0]})
     tm.assert_frame_equal(result, expected)
+
+
+@xfail_pyarrow
+@pytest.mark.parametrize(
+    "na_values, expected_result, test_id",
```
Comment (mroeschke) [inline pandas/tests/io/parser/test_na_values.py:819]: You can remove `test_id` since it's unused

File: pandas/tests/io/parser/test_na_values.py [hunk 4]
```diff
@@ -812,3 +812,32 @@ def test_bool_and_nan_to_float(all_parsers):
...
+    [
+        (
+            {"A": [-99.0, -99]},
+            DataFrame({"A": [np.nan, np.nan, np.nan, np.nan]}),
```
Comment (mroeschke) [inline pandas/tests/io/parser/test_na_values.py:823]: Could you move this to the body of the test since it's the same for both parameterized cases?

**Review by mroeschke — COMMENTED** [review]
Yeah I don't think you have to worry about those test failures. They are impacting other PRs as well

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @sshu2017

**Pull request merged**